src/main/java/Bob/Bob.java

```diff
@@ -0,0 +1,167 @@
+// */
+//public class Bob {
+//    private Storage storage;
+//    private TaskList tasks;
+//    private Ui ui;
+//    private Parser parser;
+//
+//    /**
+//     * Constructs a Bob instance with a specified file path for storage.
+//     * @param filePath The file path where tasks are stored.
+//     */
+//    public Bob(String filePath) {
+//        ui = new Ui();
+//        storage = new Storage(filePath);
+//        parser = new Parser();
+//        try {
+//            tasks = new TaskList(storage.load());
+//        } catch (BobException e) {
+//            ui.showError(e.getMessage());
+//            tasks = new TaskList();
+//        }
+//    }
+//
+//    // Constructor without argument
+//    public Bob() {
+//        ui = new Ui();
+//        storage = new Storage("./data/bob.csv");
+//        parser = new Parser();
+//        try {
+//            tasks = new TaskList(storage.load());
+//        } catch (BobException e) {
+//            ui.showError(e.getMessage());
+//            tasks = new TaskList();
+//        }
+//    }
+//
+//    /**
+//     * Starts Bob chatbot, handling user input and executing commands.
+//     */
+//    public void run() {
+//        ui.showWelcome();
+//        Scanner scanner = new Scanner(System.in);
+//        String input;
+//
+//        do {
+//            input = scanner.nextLine().trim();
+//            try {
+//                Command command = parser.parse(input);
+//                command.execute(tasks.getTasks(), storage, ui);
+//            } catch (BobException e) {
+//                ui.showError(e.getMessage());
+//            }
+//        } while (!input.equals("bye"));
+//
+//        ui.showGoodbye();
+//        scanner.close();
+//    }
+//
+//    /**
+//     * The main entry point of Bob chatbot.
+//     * @param args Command-line arguments.
+//     */
+//    public static void main(String[] args) {
+//        new Bob("./data/bob.csv").run();
+//    }
+//}
+
+package bob;
+
+import bob.Storage.Storage;
+import bob.TaskList.TaskList;
+import bob.Ui.Ui;
+import bob.Parser.Parser;
+import bob.Command.Command;
+import bob.Exception.BobException;
+
+import java.util.Scanner;
+
+/**
+ * The main class for Bob chatbot.
+ * Handles the initialization and interactions of the chatbot.
+ */
+public class Bob {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * Constructs a Bob instance with a specified file path for storage.
+     * @param filePath The file path where tasks are stored.
+     */
+    public Bob(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        parser = new Parser();
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (BobException e) {
+            ui.showError(e.getMessage());
+            tasks = new TaskList();
+        }
+    }
+
+    // Constructor without argument
+    public Bob() {
+        this("./data/bob.csv");
```

Comment:
Perhaps consider extracting `"./data/bob/csv"` into a static constant since for easier maintenance?